### ISSUE TITLE
Add a `Superseded` status

### ISSUE THREAD

**Issue by xinbenlv** [ethereum/EIPs#5265]
In RFC, after finalized and adopted, can be declared as superseded by another RFC

I wonder if we want to add `superseded-by: (eip number)` as a preamble section, and "Superseded" as a status after `Final`.

If an EIP's spec is completely out of date, and the only value is as a historical reference: implementer of clients / smart contracts shall be pointed at the new EIP number, shall be considered "superseded".

Examples I can think of:
1. an EIP-A specifying a fork at blocknum, and after that fork, the later fork comes up as EIP-B
2. a parameter was specified in EIP-C, later that same parameter was modified again.

Love to hear your thoughts

**Comment by xinbenlv** [comment]
Example: 

https://datatracker.ietf.org/doc/html/rfc822

```
"Obsoletes:  RFC #733  (NIC #41952)"

```

**Comment by MicahZoltu** [comment]
We have discussed superseded in the past and have decided against it.  The primary reason (IMO) for not including superseded is that it means someone has to *decide* which EIPs supersede which.  For example, there are dozens of token contracts that are better than ERC-20 and all of them would love to be the "official" token that supersedes ERC-20 and have EIP-20 point at them, but who would make the decision which one is the one that actually supersedes it?

Similarly, what if someone claims to supersede ERC-20 but not everyone agrees that it does and some people think ERC-20 is still the "correct" token contract?

TL;DR: it is a governance nightmare to add `superseded-by` to an EIP.

We could have a `supersedes` field on EIPs that points *backward*, but I feel like that wouldn't add much value and every token would *claim* to supersede ERC-20 and I fear there would be user confusion because people would assume that when they saw that there was some official process by which a token supersedes another, when really it is just the author saying "you should use this instead of that".

**Comment by Pandapip1** [comment]
How about the EIP authors decide when to use `superseded-by`? There is a good use case for it: see https://eips.ethereum.org/EIPS/eip-820 and https://eips.ethereum.org/EIPS/eip-1820.

**Comment by xinbenlv** [comment]
> How about the EIP authors decide when to use `superseded-by`? There is a good use case for it: see https://eips.ethereum.org/EIPS/eip-820 and https://eips.ethereum.org/EIPS/eip-1820.

I totally agree with @Pandapip1 on this. I think it should be only fit that the original author(s) to decide whether to move an EIP to superseded and add superseded-by.

**Comment by MicahZoltu** [comment]
> How about the EIP authors decide when to use `superseded-by`? There is a good use case for it: see https://eips.ethereum.org/EIPS/eip-820 and https://eips.ethereum.org/EIPS/eip-1820.

This would certainly be more acceptable to me, though part of me feels like when an EIP becomes final its "ownership" shifts over to a public good, and I am hesitant to give power to the original authors whose views may not align with the broader community just because they happened to be the ones who penned the original EIP.

For example, say I was the author of ERC-20 and then years later I create ERC-N where I create a new token standard.  Perhaps most people believe my ERC-N token standard is terrible, should I be able to put a mark on ERC-20 that basically tells people to go use ERC-N just because I happened to be the author of ERC-20?  This gives me a *huge* amount of power that I probably shouldn't have.  If ERC-N is a good idea, the community will adopt it, and they won't if it is a bad idea, independent of whether ERC-20 was popular or not.

**Comment by xinbenlv** [comment]
On Sat, Jul 16, 2022 at 05:21 Micah Zoltu ***@***.***> wrote:

> How about the EIP authors decide when to use superseded-by? There is a
> good use case for it: see https://eips.ethereum.org/EIPS/eip-820 and
> https://eips.ethereum.org/EIPS/eip-1820.
>
> This would certainly be more acceptable to me, though part of me feels
> like when an EIP becomes final its "ownership" shifts over to a public
> good, and I am hesitant to give power to the original authors whose views
> may not align with the broader community just because they happened to be
> the ones who penned the original EIP.
>
> For example, say I was the author of ERC-20 and then years later I create
> ERC-N where I create a new token standard. Perhaps most people believe my
> ERC-N token standard is terrible, should I be able to put a mark on ERC-20
> that basically tells people to go use ERC-N just because I happened to be
> the author of ERC-20? This gives me a *huge* amount of power that I
> probably shouldn't have.
>
true. How about "suggested superseding eip" instead?

If ERC-N is a good idea, the community will adopt it, and they won't if it

**Comment by MicahZoltu** [comment]
> true. How about "suggested superseding eip" instead?

It seems like this still has the same problem, where the person who happened to author some standard is in a privileged position that gives them power to shape how things progress in the future.  Adding "suggested" helps a little but not much I think.

**Comment by xinbenlv** [comment]
On Sat, Jul 16, 2022 at 05:45 Micah Zoltu ***@***.***> wrote:

> true. How about "suggested superseding eip" instead?
>
> It seems like this still has the same problem, where the person who
> happened to author some standard is in a privileged position that gives
> them power to shape how things progress in the future. Adding "suggested"
> helps a little but not much I think.
>
I think personally I feel the worry is much smaller than having a long old
deprecated EIP draining scarced implementer attention when there are new
better version. In the future maybe DAO can help how such decision are
governed, but until that happens, i lean to weigh more on future
compatibility.

The "no decision" that "no we will not add superseded-by EIP", is also a
decision it self

The "no one shall have the power to point to another EIP" is also an
implied power that Editors / EIP Stwards effectively hold the power of
governing what can be and not be on the EIP

Hance we have to make some tradeoff with these implied power dynamics and
considerations

**Comment by Pandapip1** [comment]
> It seems like this still has the same problem, where the person who happened to author some standard is in a privileged position that gives them power to shape how things progress in the future. Adding "suggested" helps a little but not much I think.

I don't see why that's much of a problem. If the old standard is better, there isn't an EIP police that will come and change your contract code. If an EIP author genuinely thinks a better alternative is available, then they are probably in the best position to correctly make that call.

**Comment by MicahZoltu** [comment]
My feeling is that we want each standard to stand on its own merit, and not be standing on the merit of the author's prior work.  Someone can make a great standard, and then 2 years later make a terrible standard.  *Most* people likely don't know which is better but they receive a breadcrumb to the new (terrible) standard when they should have gone and sought feedback from the broader community.

Imagine if today Fabian created a new token contract and updated ERC-20 to say that their new token superseded ERC-20.  *Many* developers, particularly new developers, would run off and start implementing this new token even though there are a dozen token standards out there that are all better than ERC-20 and many of which may be better than the new standard that Fabian created.

----

Another thing to consider is that allowing the addition of superseded can introduce some pathological incentive games where someone who authors a popular EIP (ERC-20, ERC-721, ERC-1155, etc.) can sell rights to the superseded field to authors of future EIPs.  Basically you just agree that you'll add a superseded field pointing at the highest bidder's EIP which can then be used for advertising a new standard.  By making EIPs immutable, we avoid this entire can of worms.

**Comment by xinbenlv** [comment]
I feel very convinced by the *pathological incentive games* part. 

I am not sure though, for who is the right one to make that decision to lock or not lock the EIP. 
I share with you that the person who holds the power to add ore reject a piece information might misuse/abuse that power. I am cautious regardless of that person is an author or editor. 

If we give author this power, someone / some govern body shall be able to check and balance it.
The same argument goes, if we give editor this power, there is also a need for check and balance.

 Maybe leave it to a DAO? In the future I imagine the EIP process be governed by a DAO.
Before the existence of such as DAO or governing body, I am not fully convinced that we are the right group to decide how to limit author's power

**Comment by xinbenlv** [comment]
I came across issue #4335 try to solve some similar problem.  tagging author @Arachnid as FYI

**Comment by MicahZoltu** [comment]
> Maybe leave it to a DAO? In the future I imagine the EIP process be governed by a DAO.

I'm pretty staunchly opposed to any sort of vote based governance because that deteriorates to tyranny of the majority or tyranny of the wealthy.  If by DAO you mean an *actually* autonomous organization then that doesn't actually solve anything because we still need to define how exactly we make decisions on what is and what is not acceptable behavior, which is not made any simpler with a DAO.

**Comment by xinbenlv** [comment]
I realize when we use DAO to mean different things. 

Lets drop the overused word "DAO". 

What I mean is: when we dont trust individual or group of authors, we shall also be very cautious of giving editors or any roles power without check and balance. 

Hence, I propose we leave this decision to a consensus group govenernce to drive the result.

**Comment by xinbenlv** [comment]
I plan to continue pursue this proposal.

**Comment by xinbenlv** [comment]
WIP

**Comment by axic** [comment]
Just for the record `Superseded` was a status and `superseded-by` was a field in the past. They were removed after a number of discussions, and have been replaced by the `withdrawal-reason` field (#4198). I don't remember the specifics, but my PR was just the result of a number of people arguing about the fields and I just wanted to close out the conversation.

**Comment by sambacha** [comment]
> Just for the record `Superseded` was a status and `superseded-by` was a field in the past. They were removed after a number of discussions, and have been replaced by the `withdrawal-reason` field (#4198). I don't remember the specifics, but my PR was just the result of a number of people arguing about the fields and I just wanted to close out the conversation.

thanks for the clarification 👍

**Comment by gcolvin** [comment]
Final EIPs cannot be superseded or withdrawn -- they are already in use and already the spec that is being followed.  They can be corrected with  errata, and new EIPs that do a better job can be written.  Whether people prefer to use the new one is a social thing we can't control.  But all of those old ERC-20 tokens need to keep on working.

**Comment by Pandapip1** [comment]
Dismissing stale bot.

**Comment by Pandapip1** [comment]
I would at least like the option to have an author-editable post-final-editable preamble field for notes after finalization.

**Comment by Pandapip1** [comment]
Dismissing stale bot.

**Comment by YaroslavPysanskyi** [comment]
> There has been no activity on this issue for 1 week. It will be closed after 3 months of inactivity.

**Comment by bumblefudge** [comment]
@gcolvin :

> Final EIPs cannot be superseded or withdrawn -- they are already in use and already the spec that is being followed. They can be corrected with errata, and new EIPs that do a better job can be written. Whether people prefer to use the new one is a social thing we can't control. But all of those old ERC-20 tokens need to keep on working.

@MicahZoltu :

> Another thing to consider is that allowing the addition of superseded can introduce some pathological incentive games

If I'm understanding [this PR](https://github.com/ethereum/EIPs/pull/4198) correctly, only pre-final EIPs can be withdrawn and use `withdrawal-reason`.  

Is it worth considering some kind of single-line frontmatter similar syntactically to `withdrawal-reason`, but with some other semantics specific to drawing extra attention to the Errata section in special cases after going to final status?  Something with a key like `postfinal-comment` or `errata-alert`?  Example values could be `Please see errata section for security considerations added after finalizing` or `The authors recommend implementers also consider EIP-XXX, see errata section for details.`  It's probably worth considering as well whether this field without constraints could become an endrun around #4335 or the no-external-link policies (i.e. `Please google 'MY-BIASED-REGISTRY' for further information` 😅 )

NB @kdenhartog @kvhnuke

**Comment by kdenhartog** [comment]
bump

**Comment by Pandapip1** [comment]
I would support a section that could be modified post-final.

**Comment by bumblefudge** [comment]
whoops time flew how can we re-open this.  seemed less urgent than repo-split/process-split so it kinda languished in backlog but i still want this :D

**Comment by Pandapip1** [comment]
We actually decided on a way to do this on the EIPIP meeting yesterday
